pandas/tests/arrays/categorical/test_missing.py

```diff
@@ -97,3 +98,26 @@ def test_fillna_array(self):
         expected = Categorical(["A", "B", "C", "B", "A"], dtype=cat.dtype)
         tm.assert_categorical_equal(result, expected)
         assert isna(cat[-1])  # didnt modify original inplace
+
+    @pytest.mark.parametrize(
+        "values, expected",
+        [
+            ([1, 2, 3], np.array([False, False, False])),
+            ([1, 2, np.nan], np.array([False, False, True])),
+            ([1, 2, np.inf], np.array([False, False, True])),
+        ],
+    )
+    def test_use_inf_as_na(self, values, expected):
+        # https://github.com/pandas-dev/pandas/issues/33594
+        with pd.option_context("mode.use_inf_as_na", True):
+            cat = Categorical(values)
```

Comment:
Should we also test this with putting the Categorical creation *outside* of the option context?

Reply:
It turns out that actually fails interestingly enough: https://github.com/pandas-dev/pandas/pull/33629#discussion_r410936224 (if you set the option after constructing the object it loses its effect)

Reply:
> What's even more strange is the value displays as NaN:

I think that is somewhat "expected", given the discussion above in https://github.com/pandas-dev/pandas/pull/33629#discussion_r410936224 (not that I like that behaviour though).

But regardless of how the categorical is created (with or without the option), I find it very strange that `isna` is then failing

Reply:
Hmm, being a bit confused. 
The Categorical.isna is failing, of course, since it just looks at -1 in the codes, and when inf is part of the categories, it is not -1 in the codes. Hence it is failing to detect it.

Reply:
@dsaxton so could you additionally test it for `pd.isna` ? In constract to Categorical.isna(), pd.isna should work even if the Categorical is created before the context, I think?

Reply:
@jorisvandenbossche Added some tests, let me know if this is roughly what you had in mind

Reply:
Yes, that looks good!